tests/unittests/net/test_dhcp.py

```diff
@@ -1284,6 +1287,51 @@ def test_dhcpcd_discovery_ib(
             ]
         )
 
+    @mock.patch("cloudinit.net.dhcp.subp.which", return_value="/sbin/dhcpcd")
+    @mock.patch("cloudinit.net.dhcp.os.killpg")
+    @mock.patch("cloudinit.net.dhcp.subp.subp")
+    @mock.patch("cloudinit.util.load_json")
+    @mock.patch("cloudinit.util.load_binary_file")
+    @mock.patch("cloudinit.util.write_file")
+    def test_dhcpcd_discovery_timeout(
+        self,
+        m_write_file,
+        m_load_file,
+        m_loadjson,
+        m_subp,
+        m_remove,
+        m_which,
+    ):
```

Comment:
Could we please add a docstring with some context? Maybe something like:
```
ensure that dhcpcd calls retry when they reach a timeout 
```

Reply:
Except that it's not retrying?

Reply:
err, yep